pulp_container/tests/functional/api/test_rbac_repo_content.py

```diff
@@ -1,215 +1,157 @@
-
-        cls.namespace_api = PulpContainerNamespacesApi(api_client)
-        cls.registry = cli.RegistryClient(cfg)
-        cls.registry.raise_if_unsupported(unittest.SkipTest, "Tests require podman/docker")
-        cls.registry_name = urlparse(cfg.get_base_url()).netloc
-
-        delete_orphans()
-        cls.user_creator = gen_user(
-            model_roles=[
-                "container.containernamespace_creator",
-                "container.containerremote_creator",
-                "container.containerrepository_creator",
-            ],
-        )
-        cls.user_creator2 = gen_user(
-            model_roles=[
-                "container.containernamespace_creator",
-                "container.containerremote_creator",
-            ]
-        )
-        cls.user_reader = gen_user(
-            model_roles=[
-                "container.containerrepository_viewer",
-                "container.containerdistribution_consumer",
-            ]
-        )
-        cls.user_reader2 = gen_user(model_roles=["container.containerrepository_viewer"])
-        cls.user_reader3 = gen_user(model_roles=["container.containerdistribution_consumer"])
-        cls.user_helpless = gen_user()
-
-        # create a push repo with user_creator
-        image_path = f"{REGISTRY_V2_REPO_PULP}:manifest_a"
-        cls._pull(image_path)
-        repo_name = "testcontent/perms"
-        local_url = "/".join([cls.registry_name, f"{repo_name}:1.0"])
-        cls._push(image_path, local_url, cls.user_creator)
-        cls.push_repository = cls.pushrepository_api.list(name=repo_name).results[0]
-
-        # create a second push repo with user_creator2
-        image_path = f"{REGISTRY_V2_REPO_PULP}:manifest_b"
-        cls._pull(image_path)
-        repo_name = "testcontent2/perms"
-        local_url = "/".join([cls.registry_name, f"{repo_name}:1.0"])
-        cls._push(image_path, local_url, cls.user_creator2)
-        cls.push_repository2 = cls.pushrepository_api.list(name=repo_name).results[0]
-
-        # sync a repo with user_creator
-        cls.repository = cls.user_creator["repository_api"].create(
-            ContainerContainerRepository(**gen_repo())
-        )
-        cls.remote = cls.user_creator["remote_api"].create(
-            gen_container_remote(upstream_name=PULP_FIXTURE_1)
-        )
-        sync_data = ContainerRepositorySyncURL(remote=cls.remote.pulp_href)
-        sync_response = cls.user_creator["repository_api"].sync(cls.repository.pulp_href, sync_data)
+def test_rbac_repository_content(
+    delete_orphans_pre,  # TODO evaluate
+    add_to_cleanup,
+    gen_object_with_cleanup,
+    gen_user,
+    registry_client,
+    local_registry,
+    container_remote_api,
+    container_repository_api,
+    container_push_repository_api,
+    container_distribution_api,
+    container_namespace_api,
+    container_tag_api,
+):
+    """Assert that certain users can list and read content."""
+
+    user_creator = gen_user(
```

Comment:
i don't seem to see where we cleanup created users?

Reply:
A clean-up procedure is already included in the fixture: https://github.com/pulp/pulp-smash/blob/7f8d1de5fb5fba221f3de98e001bc4990009eb42/pulp_smash/pulp3/pytest_plugin/__init__.py#L369.